runtime/sema/checker.go

```diff
@@ -104,15 +104,15 @@ type Checker struct {
 	Config                  *Config
 	Elaboration             *Elaboration
 	// initialized lazily. use beforeExtractor()
-	_beforeExtractor                   *BeforeExtractor
-	errors                             []error
-	functionActivations                *FunctionActivations
-	inCondition                        bool
-	allowSelfResourceFieldInvalidation bool
-	inAssignment                       bool
-	inInvocation                       bool
-	inCreate                           bool
-	isChecked                          bool
+	_beforeExtractor                 *BeforeExtractor
+	errors                           []error
+	functionActivations              *FunctionActivations
+	inCondition                      bool
+	resourceFieldInvalidationAllowed func(*ast.MemberExpression) *Member
```

Comment:
I'd run a field alignment check here, since golang doesn't optimize field ordering. having the bool before a function pointer would add a fair bit of padding

Reply:
Good point! Improved here and elsewhere in 8a955179e8972d5df05de0cdd4023808e1bde5ea